perlwin32: Add link to MS Build Tools

rwp0 opened the pull request:
I believe the "Download Center" link itself is no more current as well, or the packages are not there, or it is very hard to find if you manage to dig into and discover somehow.

MS Build tools could be used to build `perl` with standalone MSVC compiler.

### Issue
- https://github.com/Perl/perl5/issues/21068

rwp0 commented on README.win32:
```diff
@@ -135,10 +135,8 @@ These free versions of Visual C++ 2013-2022 Professional contain the same
-links to these packages has proven a pointless task because the links keep on
-changing so often.)
+These packages can be downloaded by following links to executables at
+https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist
```
The new link probably needs to be enclosed within `L<>` as previously.

rwp0 replied:
Amended to correct that updating commit title as well (added `perlwin32:` prefix for a single file change to match PR's).

rwp0 commented:
Trying to compile the latest Perl on Windows, read that article for reference looking for a compiler and other related tools.

tonycoz reviewed (approved):
This is a significant improvement.

xenu commented:
The text above the link says:

>These free versions of Visual C++ 2013-2022 Professional contain the same compilers and linkers that ship with the full versions, and also contain everything necessary to build Perl.

I think it should link to a compiler, not redist: https://visualstudio.microsoft.com/

rwp0 commented:
> The text above the link says:
> 
> >These free versions of Visual C++ 2013-2022 Professional contain the same compilers and linkers that ship with the full versions, and also contain everything necessary to build Perl.
> 
> I think it should link to a compiler, not redist: https://visualstudio.microsoft.com/

Iee, it should be put elsewhere in the document then.

I think for the standalone compiler (and other build tools) we can use the below link safely, would you agree?

https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/

xenu commented:
Windows SDK doesn't include a compiler. It used to, but it was removed in Windows 8 SDK.

To get a C/C++ compiler you need either full Visual Studio or "Build Tools for Visual Studio".

rwp0 commented:
> Windows SDK doesn't include a compiler. It used to, but it was removed in Windows 8 SDK.
> 
> To get a C/C++ compiler you need either full Visual Studio or "Build Tools for Visual Studio".

Interesting. Good to know, thanks.

Currently it would be as below:

https://visualstudio.microsoft.com/downloads/?q=build+tools#build-tools-for-visual-studio-2022

There's also a shortcut it seems although to a previous version:

https://aka.ms/build stools

tonycoz commented:
https://visualstudio.microsoft.com/visual-cpp-build-tools/

might be a better link for build tools

mauke commented on README.win32:
```diff
@@ -135,10 +135,8 @@ These free versions of Visual C++ 2013-2022 Professional contain the same
-links to these packages has proven a pointless task because the links keep on
-changing so often.)
+These packages can be downloaded by following links to executables at
+L<https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist>
```
Nit: There should be a `.` after the `L<...>` to end the sentence.

rwp0 commented:
Addressed the suggestions.

It seems Microsoft Visual Studio Community Edition is referred as Microsoft Visual C++ for some reason in the document.
Didn't bother to replace that.

Added a separate item, fixed some other links, formatting etc.

mauke commented on README.win32:
```diff
@@ -169,11 +174,11 @@ or later, and with MinGW64 64-bit 6.3.0 or later.
 To set up the build environment, from the Start Menu run
 IA-32 Visual Studio 20__ mode or Intel 64 Visual Studio 20__ mode as
-appropriate. Then run nmake as usually in that prompt box.
+appropriate. Then run C<nmake> as usually in that prompt box.
```
Shouldn't that be "as usual" (without ~~ly~~)?

Otherwise it looks good to me.

rwp0 replied:
Yeah, I saw that as well, and just skipped. Correcting now.

rwp0 commented:
Now we unfortunately have 3 separate commits due to conflicts.

Would be good if someone with git experience helped with "squashing" them into one.

rwp0 committed:
Add MS Build Tools links

Add new section on Microsoft Build Tools, improve formatting.

MSBT provides CLI compiler and other related tools (without the overhead of the Visual Studio IDE) to compile programs on Windows.

tonycoz commented on README.win32:
```diff
@@ -135,28 +135,41 @@ These free versions of Visual C++ 2013-2022 Professional contain the same
+
+This is also referred to as I<Build Tools for Visual Studio>.
+
+=item Microsoft C++ Build Tools
+
+There's also a standalone (IDE-less) version of the build tools mentioned
+above containing the MSVC compiler available for download from
+L<https://visualstudio.microsoft.com/visual-cpp-build-tools/>.
+
+This is also referred to as I<Build Tools for Visual Studio>.
```
This duplicates the section immediately above

tonycoz commented on README.win32:
```diff
@@ -135,28 +135,41 @@ These free versions of Visual C++ 2013-2022 Professional contain the same
-Perl should now build using the win32/Makefile.  You will need to edit that
-file to set CCTYPE to one of MSVC120-MSVC142 first.
+Perl should now build using the F<win32/Makefile>.  You will need to edit that
+file to set C<CCTYPE> to one of C<MSVC120>-C<MSVC142> first.
```
This can go up to `MSVC143`

rwp0 commented:
Reviews resolved.
